Sources/NIO/SelectorKqueue.swift

```diff
@@ -202,24 +204,24 @@ extension Selector: _SelectorBackendProtocol {
             throw IOError(errnoCode: EBADF, reason: "can't call whenReady for selector as it's \(self.lifecycleState).")
         }
 
-
         let timespec = Selector.toKQueueTimeSpec(strategy: strategy)
         let ready = try timespec.withUnsafeOptionalPointer { ts in
             Int(try KQueue.kevent(kq: self.selectorFD, changelist: nil, nchanges: 0, eventlist: events, nevents: Int32(eventsCapacity), timeout: ts))
         }
 
-        // start with no deregistrations happened
-        self.deregistrationsHappened = false
-        // temporary workaround to stop us delivering outdated events; possibly set in `deregister`
-        for i in 0..<ready where !self.deregistrationsHappened {
+        for i in 0..<ready {
             let ev = events[i]
             let filter = Int32(ev.filter)
+            let eventSequenceIdentifier = UInt32(Int(bitPattern: ev.udata))
```

Comment:
We can skip the intermediate `Int`.

```suggestion
            let eventSequenceIdentifier = UInt32(bitPattern: ev.udata)
```